SofaKernel/modules/SofaExplicitOdeSolver/src/SofaExplicitOdeSolver/EulerSolver.cpp

```diff
@@ -182,5 +244,83 @@ void EulerExplicitSolver::init()
     reinit();
 }
 
+void EulerExplicitSolver::addSeparateGravity(sofa::simulation::common::MechanicalOperations* mop, SReal dt)
```

Comment:
To follow the other method signatures and to be more explicit. Note that you will need to update calls to this function with
```c++
addSeparateGravity(mop, dt, sofa::core::VecDerivId::velocity());
```

```suggestion
void EulerExplicitSolver::addSeparateGravity(sofa::simulation::common::MechanicalOperations* mop, SReal dt, sofa::core::MultiVecDerivId v)
```

Reply:
I'll follow your suggestion. But I noticed that, by default without specifying any parameter, `addSeparateGravity` acts on `core::VecDerivId::velocity()`. Is it still valid for a free motion animation loop, where the new velocity corresponds to `core::VecDerivId::freeVelocity()`?

Reply:
This is exactly why I prefer not using defaults on parameters. 

Anyway, I think the velocity should be the one passed as parameter to the `EulerExplicitSolver::solve` method. Therefore, this would introduce a change of behavior when the `FreeMotionAnimationLoop` is used (before it was the `core::VecDerivId::velocity()`, now it will be `core::VecDerivId::freeVelocity()`). I think the later is the correct one (you want to apply gravity on the velocity without constraints first).

Reply:
So do you agree with 5c0e8585fbc42bcbb39e131c311c7e9695a1b51d?

Reply:
Yes I do !